web-client/src/pages/signup.js

```diff
@@ -1,7 +1,46 @@
 import '../components/signup.css';
 import '../global.css';
+import { useState } from 'react';
+import {useNavigate} from 'react-router-dom';
 
 export default function Signup(){
+    const [ inputs, setInputs ] = useState({
+        name: '',
+        email: '',
+        password: '',
+        dateOfBirth: '',
+        biography: '',
+        notificationPreference: '',
+        agreeToTerms: '',
+        subscribeToNewsletter: '',
+    });
+    const { name, email, password, dateOfBirth, biography, notificationPreference, agreeToTerms, subscribeToNewsletter } = inputs;
+    const navigate = useNavigate()
+
+    const onChange = e => {
+    setInputs({ ...inputs, [e.target.name]: e.target.value })
+    }
+
+    const onSubmit = async (e) => {
+    e.preventDefault();
+    const body = { name, email, password, dateOfBirth, biography, notificationPreference, agreeToTerms, subscribeToNewsletter };
+    try {
+       const r = await fetch('http://localhost:3001/users', {
+       method: 'POST',
+       headers: { 'Content-type': 'application/json' },
+       body: JSON.stringify(body),
+       });
+       const response = await r.json();
+       if(response){
+       navigate('/signin');
+       }
```

Comment:
When you call await fetch inside a try block, if for some reason you can't reach the server, an error will be raised and execution will go to the catch block.

Now, if you get a response from the server, it could be two:

1) All good (r.ok is true)

2) Server didnt like your request for some reason and rejected it with an error (r.ok is false)

So you can do like this:

```suggestion
       if (r.ok === false) {
         throw new Error('Server rejected the request for some reason');
       }
       
       // else we keep going (r.ok is true):
       return r.json();
     }).then((responseBody) => {
       navigate('/signin');
     });

```